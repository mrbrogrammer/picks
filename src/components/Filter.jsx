
/**
 * Filter Component
 *
 * @description
 *   A React component that renders a text input for filtering playlists or songs. It calls a callback function whenever the user types in the input, passing the current value to the parent component for filtering logic.
 *
 * @component
 *
 * @param {Object} props - The component props.
 * @param {function} props.onTextChange - Callback function called with the input value whenever the user types in the filter box.
 *
 * @returns {JSX.Element} A styled div containing a text input for filtering.
 *
 * @example
 * // Example usage in a parent component:
 * <Filter onTextChange={value => setFilterString(value)} />
 *
 * @notes
 * - The component does not manage its own state; it relies on the parent to handle the filter value.
 * - If `onTextChange` is not provided or not a function, typing in the input will throw an error.
 * - The input is not debounced; every keyup event triggers the callback.
 */
import React, { Component } from 'react';
import '../App.css';



// Inline style for the filter input
let defaultStyle = {
    color: '#fff'
};



class Filter extends Component {
    render() {
        return (
            <div style={defaultStyle}>
                <img alt=""/>
                <input type="text" onKeyUp={event => 
                    this.props.onTextChange(event.target.value)}/>
            </div>
        );
    }
}

export default Filter;
