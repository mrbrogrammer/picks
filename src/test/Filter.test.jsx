import { render } from "@testing-library/react";
import Filter from "../components/Filter";

describe("Test filter Component", () => {
    it("renders correctly", () => {
        const result = render(<Filter onTextChange={() => {}} />);
        expect(result).toMatchSnapshot();
    });
});