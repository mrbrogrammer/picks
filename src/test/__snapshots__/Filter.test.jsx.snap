// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test filter Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="color: rgb(255, 255, 255);"
      >
        <img
          alt=""
        />
        <input
          style="font-size: 20px; padding: 10px;"
          type="text"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="color: rgb(255, 255, 255);"
    >
      <img
        alt=""
      />
      <input
        style="font-size: 20px; padding: 10px;"
        type="text"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
